### PR TITLE
Add telemetry and icon

### PR DESCRIPTION
**Pull request by yankunhuang-pku**
* Add telemetry
* Add icon
* Update package information
* Check if the operating system is supported.
* Check if result folder or template folder exists

### CODE REVIEW

**Review by ACMoretxj — APPROVED** [review]

File: FHIR-Converter/client/src/core/common/errors/error-handler.ts [hunk 1]
```diff
@@ -7,11 +7,15 @@ import * as vscode from 'vscode';
...
+	let errorType = '';
 	if (error instanceof ConfigurationError || error instanceof ConversionError) {
-		vscode.window.showErrorMessage(localize(error.name, error.message));
+		errorType = error.name;
 	} else {
-		vscode.window.showErrorMessage(localize('error.unexpected', error.message));
+		errorType = 'error.unexpected';
 	}
+	vscode.window.showErrorMessage(localize(errorType, error.message));
+	reporter.sendTelemetryErrorEvent('error', { type: errorType, command: commandFunc.name, message: error.message, errorstack: error.stack }, {}, []);
```
Comment (qiwjin) [inline FHIR-Converter/client/src/core/common/errors/error-handler.ts:20]: let's disable error report first.

File: FHIR-Converter/client/src/view/common/commands/command-handler.ts [hunk 2]
```diff
@@ -19,6 +28,6 @@ export async function commandHandler(event) {
 		await this(event);
 	} catch (error) {
 		// Handle the error
-		errorHandler.handle(error);
+		errorHandler.handle(error, this);
```
Comment (qiwjin) [inline FHIR-Converter/client/src/view/common/commands/command-handler.ts:31]: why do we need add 'this' here? just wrap the error into ConversionError or CommandError?

File: FHIR-Converter/client/src/view/common/commands/command-handler.ts [hunk 3]
```diff
@@ -6,11 +6,20 @@
...
+import { ConversionError } from '../../../core/common/errors/conversion-error';
 import { ConfigurationError } from '../../../core/common/errors/configuration-error';
 import * as configurationConstants from '../../../core/common/constants/workspace-configuration';
+import { reporter } from '../../../telemetry/telemetry';
+import * as osUtils from '../../../core/common/utils/os-utils';
 
 export async function commandHandler(event) {
 	try {
+		// Telemetry for commands
+		reporter.sendTelemetryEvent('command', { command: this.name });
```
Comment (qiwjin) [inline FHIR-Converter/client/src/view/common/commands/command-handler.ts:18]: what's the type of 'this'?

File: FHIR-Converter/client/src/view/common/commands/command-handler.ts (hunk 2 above)
Comment (yankunhuang-pku) [reply FHIR-Converter/client/src/view/common/commands/command-handler.ts:31]: In order to record which command triggered the error.

File: FHIR-Converter/client/src/view/common/commands/command-handler.ts (hunk 3 above)
Comment (yankunhuang-pku) [reply FHIR-Converter/client/src/view/common/commands/command-handler.ts:18]: function

File: FHIR-Converter/client/src/view/user-commands/convert.ts [hunk 4]
```diff
@@ -15,52 +15,64 @@ import { showResultEditor } from '../common/editor/show-result-editor';
...
+			if (history.length === 1) {
+				// Show result in the editor
+				await showResultEditor(vscode.Uri.file(history[0]));
+			} else if (history.length > 1) {
+				// Show result with the differential view
+				await showDifferentialView(vscode.Uri.file(history[1]), vscode.Uri.file(history[0]));
+			}
 		}
+		await vscode.commands.executeCommand('workbench.action.closeOtherEditors');
+	} catch (err) {
```
Comment (qiwjin) [inline FHIR-Converter/client/src/view/user-commands/convert.ts:72]: does it possible don't catch this err? just use try/finally

**Review by qiwjin — APPROVED** [review]

File: FHIR-Converter/client/src/view/user-commands/convert.ts (hunk 4 above)
Comment (yankunhuang-pku) [reply FHIR-Converter/client/src/view/user-commands/convert.ts]: Thanks, updated it.

**Review by qiwjin — APPROVED** [review]

**Pull request merged**